client/v3/client_test.go

```diff
@@ -266,6 +268,105 @@ func TestSyncFiltersMembers(t *testing.T) {
 	}
 }
 
+func TestClientRejectOldCluster(t *testing.T) {
+	testutil.BeforeTest(t)
+	var tests = []struct {
+		name          string
+		endpoints     []string
+		versions      []string
+		expectedError error
+	}{
+		{
+			name:          "all new versions with the same value",
+			endpoints:     []string{"192.168.3.41:22379", "192.168.3.41:22479", "192.168.3.41:22579"},
+			versions:      []string{"3.5.4", "3.5.4", "3.5.4"},
+			expectedError: nil,
+		},
+		{
+			name:          "all new versions with different values",
+			endpoints:     []string{"192.168.3.41:22379", "192.168.3.41:22479", "192.168.3.41:22579"},
+			versions:      []string{"3.5.4", "3.5.4", "3.4.0"},
+			expectedError: nil,
+		},
+		{
+			name:          "all old versions with different values",
+			endpoints:     []string{"192.168.3.41:22379", "192.168.3.41:22479", "192.168.3.41:22579"},
+			versions:      []string{"3.3.0", "3.3.0", "3.4.0"},
+			expectedError: ErrOldCluster,
+		},
+		{
+			name:          "all old versions with the same value",
+			endpoints:     []string{"192.168.3.41:22379", "192.168.3.41:22479", "192.168.3.41:22579"},
+			versions:      []string{"3.3.0", "3.3.0", "3.3.0"},
+			expectedError: ErrOldCluster,
+		},
+	}
+
+	for i, tt := range tests {
```

Comment:
Please use golang subtest something like below,
```
	for _, tc := range cases {
		t.Run(tc.name, func(t *testing.T) {
                    ......
		})
	}
```

Reply:
Using `t.Run` will make it a separate test, which means no need to use `i` in logging to identify test case.

Reply:
I have modified.